[UnifiedPDF] Re-enable the PDF HUD

hortont424 opened the pull request:
#### 77b692aefd85b2b8896817c2198f1f6cec4e0042
<pre>
[UnifiedPDF] Re-enable the PDF HUD
<a href="https://bugs.webkit.org/show_bug.cgi?id=264608">https://bugs.webkit.org/show_bug.cgi?id=264608</a>
&lt;<a href="https://rdar.apple.com/problem/118154838">rdar://problem/118154838</a>&gt;

Reviewed by Richard Robinson and Simon Fraser.

Hoist PDFPlugin&apos;s HUD creation/destruction code into PDFPluginBase, and add
empty implementations of the actions.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::PDFPlugin):
(WebKit::PDFPlugin::teardown):
(WebKit::PDFPlugin::~PDFPlugin): Deleted.
(WebKit::PDFPlugin::hudEnabled const): Deleted.
(WebKit::PDFPlugin::frameForHUD const): Deleted.
(WebKit::PDFPlugin::updatePDFHUDLocation): Deleted.
(WebKit::PDFPlugin::convertFromPluginToPDFView const): Deleted.
(WebKit::PDFPlugin::convertFromPDFViewToRootView const): Deleted.
(WebKit::PDFPlugin::convertFromRootViewToPDFView const): Deleted.
(WebKit::PDFPlugin::convertFromPDFViewToScreen const): Deleted.
(WebKit::PDFPlugin::boundsOnScreen const): Deleted.
(WebKit::PDFPlugin::visibilityDidChange): Deleted.
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
(WebKit::PDFPluginBase::identifier const):
(WebKit::PDFPluginBase::visibilityDidChange): Deleted.
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
(WebKit::PDFPluginBase::PDFPluginBase):
(WebKit::PDFPluginBase::~PDFPluginBase):
(WebKit::PDFPluginBase::teardown):
(WebKit::PDFPluginBase::geometryDidChange):
(WebKit::PDFPluginBase::visibilityDidChange):
(WebKit::PDFPluginBase::convertFromPluginToPDFView const):
(WebKit::PDFPluginBase::convertFromPDFViewToRootView const):
(WebKit::PDFPluginBase::convertFromRootViewToPDFView const):
(WebKit::PDFPluginBase::convertFromPDFViewToScreen const):
(WebKit::PDFPluginBase::boundsOnScreen const):
(WebKit::PDFPluginBase::updatePDFHUDLocation):
(WebKit::PDFPluginBase::frameForHUD const):
(WebKit::PDFPluginBase::hudEnabled const):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::teardown):
(WebKit::UnifiedPDFPlugin::countFindMatches):
(WebKit::UnifiedPDFPlugin::findString):
(WebKit::UnifiedPDFPlugin::zoomIn):
(WebKit::UnifiedPDFPlugin::zoomOut):
(WebKit::UnifiedPDFPlugin::save):
(WebKit::UnifiedPDFPlugin::openWithPreview):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::createPDFHUD):
(WebKit::WebPage::updatePDFHUDLocation):
(WebKit::WebPage::removePDFHUD):

Canonical link: <a href="https://commits.webkit.org/270560@main">https://commits.webkit.org/270560@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/33b12963ab51cc328c6d7506ac8dd8b5946ba84c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25770 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4377 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27054 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/27868 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/23601 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6136 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/1812 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/23711 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26020 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3285 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22208 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28448 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/2908 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/29233 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/23519 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23534 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27098 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/2925 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1158 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4312 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6196 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3378 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3241 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
```diff
@@ -208,6 +242,53 @@
+
+IntRect PDFPluginBase::boundsOnScreen() const
+{
+    return WebCore::Accessibility::retrieveValueFromMainThread<WebCore::IntRect>([&] () -> WebCore::IntRect {
```
does it need to capture self weakly? or is this method sync?

hortont424 replied:
`callOnMainThreadAndWait` so it's sync.

hortont424 committed:
[UnifiedPDF] Re-enable the PDF HUD
https://bugs.webkit.org/show_bug.cgi?id=264608
<rdar://problem/118154838>

Reviewed by Richard Robinson and Simon Fraser.

Hoist PDFPlugin's HUD creation/destruction code into PDFPluginBase, and add
empty implementations of the actions.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::PDFPlugin):
(WebKit::PDFPlugin::teardown):
(WebKit::PDFPlugin::~PDFPlugin): Deleted.
(WebKit::PDFPlugin::hudEnabled const): Deleted.
(WebKit::PDFPlugin::frameForHUD const): Deleted.
(WebKit::PDFPlugin::updatePDFHUDLocation): Deleted.
(WebKit::PDFPlugin::convertFromPluginToPDFView const): Deleted.
(WebKit::PDFPlugin::convertFromPDFViewToRootView const): Deleted.
(WebKit::PDFPlugin::convertFromRootViewToPDFView const): Deleted.
(WebKit::PDFPlugin::convertFromPDFViewToScreen const): Deleted.
(WebKit::PDFPlugin::boundsOnScreen const): Deleted.
(WebKit::PDFPlugin::visibilityDidChange): Deleted.
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
(WebKit::PDFPluginBase::identifier const):
(WebKit::PDFPluginBase::visibilityDidChange): Deleted.
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
(WebKit::PDFPluginBase::PDFPluginBase):
(WebKit::PDFPluginBase::~PDFPluginBase):
(WebKit::PDFPluginBase::teardown):
(WebKit::PDFPluginBase::geometryDidChange):
(WebKit::PDFPluginBase::visibilityDidChange):
(WebKit::PDFPluginBase::convertFromPluginToPDFView const):
(WebKit::PDFPluginBase::convertFromPDFViewToRootView const):
(WebKit::PDFPluginBase::convertFromRootViewToPDFView const):
(WebKit::PDFPluginBase::convertFromPDFViewToScreen const):
(WebKit::PDFPluginBase::boundsOnScreen const):
(WebKit::PDFPluginBase::updatePDFHUDLocation):
(WebKit::PDFPluginBase::frameForHUD const):
(WebKit::PDFPluginBase::hudEnabled const):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::teardown):
(WebKit::UnifiedPDFPlugin::countFindMatches):
(WebKit::UnifiedPDFPlugin::findString):
(WebKit::UnifiedPDFPlugin::zoomIn):
(WebKit::UnifiedPDFPlugin::zoomOut):
(WebKit::UnifiedPDFPlugin::save):
(WebKit::UnifiedPDFPlugin::openWithPreview):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::createPDFHUD):
(WebKit::WebPage::updatePDFHUDLocation):
(WebKit::WebPage::removePDFHUD):

Canonical link: https://commits.webkit.org/270560@main
